File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditDescriptionEventDto.java
package de.helfenkannjeder.helfomat.api.organization.event;

import de.helfenkannjeder.helfomat.core.organization.OrganizationId;

/**
 * @author Valentin Zickner
 */
public class OrganizationEditDescriptionEventDto extends OrganizationEventDto {
    private String description;

    OrganizationEditDescriptionEventDto() {
    }

    public OrganizationEditDescriptionEventDto(OrganizationId organizationId, String description) {
        super(organizationId);
        this.description = description;
    }

    public String getDescription() {
        return description;
    }

    @Override
    public <T> T visit(OrganizationEventDtoVisitor<T> visitor) {
        return visitor.visit(this);
    }

}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditDescriptionEventDto.kt
package de.helfenkannjeder.helfomat.api.organization.event

import de.helfenkannjeder.helfomat.core.organization.OrganizationId

/**
 * @author Valentin Zickner
 */
data class OrganizationEditDescriptionEventDto(
    override val organizationId: OrganizationId,
    val description: String?,
    override val eventApplicable: EventApplicability
) : OrganizationEventDto {

    override fun <T> visit(visitor: OrganizationEventDtoVisitor<T>): T {
        return visitor.visit(this)
    }

}